pkg/sqlite/graphloader.go

```diff
@@ -0,0 +1,127 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/operator-framework/operator-registry/pkg/registry"
+)
+
+type SQLGraphLoader struct {
+	Querier     registry.Query
+	PackageName string
+}
+
+func NewSQLGraphLoader(dbFilename, name string) (*SQLGraphLoader, error) {
+	querier, err := NewSQLLiteQuerier(dbFilename)
+	if err != nil {
+		return nil, err
+	}
+
+	return &SQLGraphLoader{
+		Querier:     querier,
+		PackageName: name,
+	}, nil
+}
+
+func NewSQLGraphLoaderFromDB(db *sql.DB, name string) (*SQLGraphLoader, error) {
+	return &SQLGraphLoader{
+		Querier:     NewSQLLiteQuerierFromDb(db),
+		PackageName: name,
+	}, nil
+}
+
+func (g *SQLGraphLoader) Generate() (*registry.Package, error) {
+	ctx := context.TODO()
+	defaultChannel, err := g.Querier.GetDefaultPackage(ctx, g.PackageName)
+	if err != nil {
+		return nil, err
+	}
+
+	channelEntries, err := g.Querier.GetChannelEntriesFromPackage(ctx, g.PackageName)
+	if err != nil {
+		return nil, err
+	}
+
+	channels, err := g.GraphFromEntries(channelEntries)
+	if err != nil {
+		return nil, err
+	}
+
+	return &registry.Package{
+		Name:           g.PackageName,
+		DefaultChannel: defaultChannel,
+		Channels:       channels,
+	}, nil
+}
+
+// GraphFromEntries builds the graph from a set of channel entries
+func (g *SQLGraphLoader) GraphFromEntries(channelEntries []registry.ChannelEntryAnnotated) (map[string]registry.Channel, error) {
```

Comment:
suggestion: can we make this a function instead of a method of the `SQLGraphLoader`? the function doesn't use any of the `SQLGraphLoader` data. This makes the func easier to use in other pkgs

Reply:
seems reasonable